### PR TITLE
fix/condo/doma-3698/wrong-roles-sorting

### PR DESCRIPTION
**Pull request by AleX83Xpert**
Custom solution: Sorting of roles names (LocalizedText) at the front.

### CODE REVIEW

File: apps/condo/domains/contact/components/contactRoles/ContactRolesSettingsContent.tsx [hunk 1]
```diff
@@ -63,6 +66,24 @@ export const ContactRolesSettingsContent = (props) => {
...
 
+    const sortedRoles = useMemo(() => {
+        const sorterByName = sorters.reduce((sorter, next) => {
+            if (next.columnKey === 'name') {
+                return next
+            }
+            return null
+        }, null)
+
+        if (sorterByName && roles.length > 0) {
```
Comment (akarjakin) [inline apps/condo/domains/contact/components/contactRoles/ContactRolesSettingsContent.tsx:77]: `if (!isEmpty(sorterByName) && !isEmpty(roles)) {`

File: apps/condo/domains/contact/components/contactRoles/ContactRolesSettingsContent.tsx [hunk 2]
```diff
@@ -63,6 +66,24 @@ export const ContactRolesSettingsContent = (props) => {
...
+            const isDesc = sorterByName.order === 'descend'
+            const ascSorted = _sortBy(roles, ({ name }) => lowerCase(name))
```
Comment (akarjakin) [inline apps/condo/domains/contact/components/contactRoles/ContactRolesSettingsContent.tsx:79]: `_.orderBy` can do DESC sorting :)

File: apps/condo/domains/contact/components/contactRoles/ContactRolesSettingsContent.tsx (hunk 1 above)
Comment (AleX83Xpert) [reply apps/condo/domains/contact/components/contactRoles/ContactRolesSettingsContent.tsx:77]: `sorterByName` is always null or an object, see reducer function.

`roles` is always an array.
image

File: apps/condo/domains/contact/components/contactRoles/ContactRolesSettingsContent.tsx (hunk 2 above)
Comment (AleX83Xpert) [reply apps/condo/domains/contact/components/contactRoles/ContactRolesSettingsContent.tsx]: Refactored :)

**Review by SavelevMatthew — APPROVED** [review]

**Review by DanilovMaxim13 — APPROVED** [review]

**Pull request merged**